src/main/java/com/linecorp/armeria/server/http/tomcat/TomcatServiceInvocationHandler.java

```diff
@@ -335,6 +101,11 @@ public long getBytesWritten() {
 
             ServiceInvocationContext.setCurrent(ctx);
             try {
+                final Adapter coyoteAdapter = connector.getProtocolHandler().getAdapter();
+                if (coyoteAdapter == null) {
+                    // Tomcat is not configured / stopped.
+                    throw new ServiceUnavailableException();
```

Comment:
Could we extract this check into a new method and perform the same check before at the very beginning of `invoke()`, so we do not waste any resources when Tomcat is not up?